File: examples/python-examples/SignedDistance.dox.py
from meshlib import mrmeshpy

mesh1 = mrmeshpy.loadMesh("mesh1.ctm")
mesh2 = mrmeshpy.loadMesh("mesh2.ctm")

a = mrmeshpy.findSignedDistance(mesh1, mesh2)
print(f"Signed distance between meshes is {a.signedDist}")

File: examples/cpp-examples/SignedDistance.dox.cpp
#include "MRMesh/MRMeshLoad.h"
#include "MRMesh/MRMesh.h"
#include "MRMesh/MRMeshMeshDistance.h"
#include <iostream>

int main()
{
    auto mesh1Res = MR::MeshLoad::fromAnySupportedFormat( "mesh1.ctm" );
    if ( !mesh1Res.has_value() )
    {
        std::cerr << mesh1Res.error();
        return 1;
    }
    auto mesh2Res = MR::MeshLoad::fromAnySupportedFormat( "mesh2.ctm" );
    if ( !mesh2Res.has_value() )
    {
        std::cerr << mesh2Res.error();
        return 1;
    }

    auto dist = MR::findSignedDistance( *mesh1Res, *mesh2Res );
    std::cout << "Signed distance between meshes is " << dist.signedDist << "\n";
    return 0;
}
